_data-prepper/pipelines/configuration/processors/anomaly-detector.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: default
+title: Anomaly detector processor
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# anomaly_detector
+
+## Overview
+
+The anomaly detector processor takes structured data and runs anomaly detection algorithms on fields that you can configure in that data. The data must be either an integer or a real number for the anomaly detection algorithm to detect anomalies. Deploying the aggregate processor in a pipeline before the anomaly detector processor can help you achieve the best results, as the aggregate processor automatically aggregates events by key and keeps them on the same host. For example, if you are searching for an anomaly in latencies from a specific IP address and if all the events go to the same host, then the host has more data for these events. This additional data results in better training of the machine learning (ML) algorithm, which results in better anomaly detection. 
+
+## Configuration
+
+You can configure the anomaly detector processor by specifying a key and the options for the selected mode. You can use the following options to configure the anomaly detector processor.
+
+| Name | Required | Description |
+| :--- | :--- | :--- |
+| `keys` | Yes | A non-ordered `List<String>` that is used as input to the ML algorithm to detect anomalies in the values of the keys in the list. At least one key is required.
+| `mode` | Yes |  The ML algorithm (or model) used to detect anomalies. You must provide a mode. See [random_cut_forest mode](#random_cut_forest-mode).
+
+### Keys
+
+Keys that are used in the anomaly detector processor are present in the input event. For example, if the input event is `{"key1":value1, "key2":value2, "key3":value3}`, then any of the keys (such as `key1`, `key2`, `key3`) in that input event can be used as anomaly detector keys as long as their value (such as `value1`, `value2`, `value3`) is an integer or real number.
+
+### random_cut_forest mode
+
+The random cut forest (RCF) ML algorithm is an unsupervised algorithm for detecting anomalous data points within a dataset. To detect anomalies, the anomaly detector processor uses the `random_cut_forest` mode.
+
+| Name | Description |
+| :--- | :--- |
+| `random_cut_forest` | Processes events using the RCF ML algorithm to detect anomalies. After passing a group of events with latency, a value between 0.2 and 0.3 is passed through the anomaly detector processor. When an event with latency value 11.5 is sent, the following anomaly event is generated.| 
```

Comment:
I'd prefer that we not introduce within the table what follows it. Do we need to include the last sentence, or is line 36 sufficient?

Reply:
@natebower The example contains all of the information in this sentence, so it's safe to remove it.